### DISCUSSION TITLE
Questions

### DISCUSSION THREAD

**Discussion by MJia1202** [do-mpc/do-mpc#394] [Q&A]
Hi!

I'm exploring the possibility of using the NMPC solver in parallel in a deep learning project using Pytorch, is the solver thread safe? Has anyone tested the feature? Thanks for the help!

**Comment by 4flixt** [comment]
Hi @MJia1202,

do-mpc is a tool to build optimal control problems from dynamic models using CasADi as its backend. Your intended application is to vague to really help you with your question. I recommend you check the documentation and the examples to get an idea of the suggested use-cases. This will probably help you to understand if the tool might be useful for you.

Best,
4flixt.
